geode-for-redis/src/integrationTest/java/org/apache/geode/redis/internal/commands/executor/set/AbstractSScanIntegrationTest.java

```diff
@@ -54,295 +74,397 @@ public void tearDown() {
+    final int iterationCount = 500;
+    Jedis jedis1 = jedis.getConnectionFromSlot(SLOT_FOR_KEY);
+    Jedis jedis2 = jedis.getConnectionFromSlot(SLOT_FOR_KEY);
+
+    new ConcurrentLoopingThreads(iterationCount,
+        (i) -> multipleSScanAndAssertOnContentOfResultSet(i, jedis1, initialMemberData),
+        (i) -> multipleSScanAndAssertOnContentOfResultSet(i, jedis2, initialMemberData))
+            .run();
+    assertThat(jedis.smembers(KEY)).containsExactlyInAnyOrderElementsOf(initialMemberData);
+
+    jedis1.close();
+    jedis2.close();
+  }
 
-    String cursor = "-100";
+  private void multipleSScanAndAssertOnContentOfResultSet(int iteration, Jedis jedis,
+      final Set<String> initialMemberData) {
+
+    List<String> allEntries = new ArrayList<>();
     ScanResult<String> result;
+    String cursor = ZERO_CURSOR;
+
     do {
-      result = jedis.sscan("b", cursor);
-      allEntries.addAll(result.getResult());
+      result = jedis.sscan(KEY, cursor);
       cursor = result.getCursor();
+      List<String> resultEntries = result.getResult();
+      allEntries.addAll(resultEntries);
     } while (!result.isCompleteIteration());
 
-    assertThat(allEntries).containsExactlyInAnyOrder("green", "orange", "yellow");
+    assertThat(allEntries).as("failed on iteration " + iteration)
+        .containsAll(initialMemberData);
   }
 
-  @Test
-  public void givenCursorGreaterThanUnsignedLongCapacity_returnsCursorError() {
-    assertThatThrownBy(() -> jedis.sscan("a", "18446744073709551616"))
-        .hasMessageContaining(ERROR_CURSOR);
+  @SuppressWarnings("unchecked")
+  private ScanResult<byte[]> sendCustomSscanCommand(String key, String... args) {
+    List<Object> result = (List<Object>) (jedis.sendCommand(key, Protocol.Command.SSCAN, args));
+    return new ScanResult<>((byte[]) result.get(0), (List<byte[]>) result.get(1));
   }
 
-  @Test
-  public void givenNegativeCursorGreaterThanUnsignedLongCapacity_returnsCursorError() {
-    assertThatThrownBy(() -> jedis.sscan("a", "-18446744073709551616"))
-        .hasMessageContaining(ERROR_CURSOR);
+  private void initializeThreeMemberSet() {
+    jedis.sadd(KEY, FIELD_ONE, FIELD_TWO, FIELD_THREE);
   }
 
-  @Test
-  public void givenInvalidRegexSyntax_returnsEmptyArray() {
-    jedis.sadd("a", "1");
-    ScanParams scanParams = new ScanParams();
-    scanParams.count(1);
-    scanParams.match("\\p");
+  private Set<byte[]> initializeThreeMemberByteSet() {
+    Set<byte[]> set = new HashSet<>();
+    set.add(FIELD_ONE.getBytes());
+    set.add(FIELD_TWO.getBytes());
+    set.add(FIELD_THREE.getBytes());
+    jedis.sadd(KEY.getBytes(), FIELD_ONE.getBytes(), FIELD_TWO.getBytes(), FIELD_THREE.getBytes());
+    return set;
+  }
 
-    ScanResult<byte[]> result =
-        jedis.sscan("a".getBytes(), "0".getBytes(), scanParams);
+  private Set<String> initializeThousandMemberSet() {
+    Set<String> set = new HashSet<>();
+    int SIZE_OF_SET = 1000;
```

Comment:
This variable, and the one in the method below, should not be in all-caps, as that is reserved for static final constants. The [Geode Code Style Guide page](https://cwiki.apache.org/confluence/display/GEODE/Code+Style+Guide) on the wiki contains a link to the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html) which details the conventions we (try to) follow when writing code for Geode.